devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -548,12 +544,8 @@ private void triggerCompile(boolean test, boolean prepareNeeded) throws MojoExec
             triggerPrepare(test);
         }
 
-        // compile the Kotlin sources if needed
-        executeIfConfigured(ORG_JETBRAINS_KOTLIN, KOTLIN_MAVEN_PLUGIN, test ? "test-compile" : "compile",
-                Collections.emptyMap());
-
         // Compile the Java sources if needed
-        executeIfConfigured(ORG_APACHE_MAVEN_PLUGINS, MAVEN_COMPILER_PLUGIN, test ? "testCompile" : "compile",
+        executeIfConfigured(ORG_APACHE_MAVEN_PLUGINS, MAVEN_COMPILER_PLUGIN, test ? "process-test-classes" : "process-classes",
```

Comment:
Maybe it's just too late, but I'm not getting this change: `maven-compiler-plugin` does _not_ have a `process-test-classes` or `process-classes` goal/mojo!

Also, I don't see how the original problem with `jandex-maven-plugin` is solved in this PR.
For that we'd need to "cherry-pick" yet another plugin execution (the jandex one).

I'm now back to my idea from some while ago that we should not contrinue cherry-picking like that but try to gather all executions that are bound to the respective phases and execute all of them.
This would be a kind of more flexible, but also more eloborate `@Execute`.

Anyway, I think that new approach should be a separate PR.

Reply:
I think the main goal here is to call the `process-classes` phase instead of calling plugin specific goals, this would avoid having too detect whether or not this is a kotlin project, or if the jandex plugin is enabled. 

I agree that my code is not correct as it should not rely on the `maven-compiler-plugin` but on maven `phase` which (from my point of view) is similar to the `@Execute`.

Reply:
Thing is that we currently don't execute entire _phase(s)_ in `DevMojo` (which `@Execute` would).
Instead, we _assume_ that if the user added `compile` before `quarkus:dev`, the `maven-compiler-plugin` was executed with its default goal bindings (e.g. `compile` goal bound to `compile` phase). This is only wrong in rare usecases, e.g. when the compiler plugin is skiped via config or so.
That's also why we "cherry-pick" the compile execution if there was _no_ phase like `compile` or a _lower_ one like `validate`.